ports/mimxrt/mimxrt_flash.c

```diff
@@ -161,7 +102,7 @@ STATIC mp_obj_t mimxrt_flash_writeblocks(size_t n_args, const mp_obj_t *args) {
     uint32_t offset = mp_obj_get_int(args[1]) * SECTOR_SIZE_BYTES;
 
     if (n_args == 3) {
-        status = flash_erase_block(self->flash_base + offset);
+        status = flash_erase_sector(self->flash_base + offset);
```

Comment:
Why has this changed from a block to sector?

Reply:
The previous implementation of `flash_erase_block` was misleading.

```
status_t flash_erase_block(uint32_t erase_addr) __attribute__((section(".ram_functions")));
status_t flash_erase_block(uint32_t erase_addr) {
    status_t status;
    SCB_CleanInvalidateDCache();
    SCB_DisableDCache();
    __disable_irq();
    status = flexspi_nor_flash_erase_sector(BOARD_FLEX_SPI, erase_addr);
    __enable_irq();
    SCB_EnableDCache();
    return status;
}
```
As it used the `flexspi_nor_flash_erase_sector` function. As described in the documentation of the flash used with the MIMXRT there is a distinction between `sector` and `block` with `block` being of larger size.

For instance the `IS25LP064A` has a sector size of 4Kbyte and blocks of either 32 or 64 Kbyte (depending on the operation used).